### DISCUSSION TITLE
Using LuaLsp addons with Neovim?

### DISCUSSION THREAD

**Discussion by bauripalash** [neovim/neovim#27771] [Q&A]
I need to use some lua Isp addons with neovim, but I couldn't find any guides or docs on how to do that.
How do I use LuaLsp addons with neovim and Isp-config?

**Comment by justinmk** [comment]
No idea what you're asking. It helps to provide links or a references. First question is what is "LuaLsp"?

**Reply by bauripalash** [reply]
https://github.com/neovim/nvim-lspconfig/blob/master/doc/server_configurations.md#lua_ls

This one

**Reply by Phantasimay** [reply]
do you mean you want to instal lua language server? 
you can install it via [mason](https://github.com/williamboman/mason.nvim)

**Reply by bauripalash** [reply]
> do you mean you want to instal lua language server? 
> you can install it via [mason](https://github.com/williamboman/mason.nvim)

@Phantasimay , No, I meant the addons  lua language server provides, like these https://luals.github.io/wiki/addons/

**Reply by Phantasimay** [reply]
after i read your reference, i hope this will help you:

```lua

require 'lspconfig'.lua_ls.setup({
	-- ... other configs
	settings = {
		Lua = {
			diagnostics = {
				globals = { 'vim' }
			},
			runtime = {
				version = "LuaJIT",
				-- path = vim.split(package.path, ";"),
			},
			completion = { callSnippet = "Replace" },
			telemetry = { enable = false },
			workspace = {
				userThirdParty = false, -- change to true for enable addon
                                checkThirdParty = false,
				library = {                           -- your lua addon directory
					"${3rd}/luv/library",
					unpack(vim.api.nvim_get_runtime_file("", true)),
					-- [vim.fn.expand("$VIMRUNTIME/lua")] = true,
					-- [vim.fn.expand("$VIMRUNTIME/lua/vim/lsp")] = true,
				},
			},
		}
	}
})

```

Tips:
See lua table as JSON table help me to understand lua
Unfortunatelly we still must download LSP Addon manualy, i hope someday mason support addon downloader 
@bauripalash
